Libraries/NavigationExperimental/NavigationTypeDefinition.js

```diff
@@ -111,4 +113,25 @@ export type NavigationStyleInterpolator = (
   props: NavigationSceneRendererProps,
 ) => Object;
 
-export type NavigationTransitionConfigurator = () => NavigationTransitionSpec;
+export type NavigationTransitionConfigurator = (
+  transitionProps: NavigationTransitionProps,
+  prevTransitionProps: ?NavigationTransitionProps,
+) => NavigationTransitionSpec;
+
+export type AnimationConfig = Object;
+
+export type AnimationEndResult = {
+  finished: bool
+};
+
+export type AnimationEndCallback = (result: AnimationEndResult) => void;
+
+export type CompositeAnimation = {
+  start: (callback?: ?AnimationEndCallback) => void;
+  stop: () => void;
+};
+
+export type NavigationTransitionTiming = (
+  value: NavigationAnimatedValue,
+  config: AnimationConfig,
+) => CompositeAnimation;
```

Comment:
I agree. For now, let's keep them as Object and hopefully flow won't
complain about it, and hopefully we won't need `NavigationAnimatedValue`
soon.

Hedge

On Thu, Jun 23, 2016 at 6:49 PM, Jake Murzy <notifications@github.com>
wrote:

> In Libraries/NavigationExperimental/NavigationTypeDefinition.js
> <https://github.com/facebook/react-native/pull/8306#discussion_r68343711>:
>
> > +
> > +export type AnimationEndResult = {
> > +  finished: bool
> > +};
> > +
> > +export type AnimationEndCallback = (result: AnimationEndResult) => void;
> > +
> > +export type CompositeAnimation = {
> > +  start: (callback?: ?AnimationEndCallback) => void;
> > +  stop: () => void;
> > +};
> > +
> > +export type NavigationTransitionTiming = (
> > +  value: NavigationAnimatedValue,
> > +  config: AnimationConfig,
> > +) => CompositeAnimation;
>
> I totally agree. 👍 Updated the PR.
>
> Should we remove NavigationAnimatedValue as well? Perhaps the best way to
> approach this is to expose the type definitions from Animated itself so
> we can just import them from there. I might submit another PR to fix this
> in Animated.
>
> 🍺
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/react-native/pull/8306/files/86576fc702e3785a6218736ea6b8b55df5d90f9a#r68343711>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABb0t6M9t9bMyx4VftX4Oh4DZ_OLNVjoks5qOzeqgaJpZM4I7gG->
> .
>